server/models/Order.ts

```diff
@@ -1,312 +1,478 @@
+  platformTipEligible?: boolean;
+  AccountingCategoryId?: number;
 }
 
+export type OrderPk = 'id';
+export type OrderId = Order[OrderPk];
+export type OrderOptionalAttributes =
+  | 'id'
+  | 'CreatedByUserId'
+  | 'CollectiveId'
+  | 'currency'
+  | 'totalAmount'
+  | 'description'
+  | 'SubscriptionId'
+  | 'createdAt'
+  | 'updatedAt'
+  | 'deletedAt'
+  | 'PaymentMethodId'
+  | 'processedAt'
+  | 'privateMessage'
+  | 'TierId'
+  | 'FromCollectiveId'
+  | 'publicMessage'
+  | 'quantity'
+  | 'status'
+  | 'data'
+  | 'taxAmount'
+  | 'interval'
+  | 'tags'
+  | 'platformTipAmount'
+  | 'platformTipEligible'
+  | 'AccountingCategoryId';
+export type OrderCreationAttributes = Optional<OrderAttributes, OrderOptionalAttributes>;
+
 export type OrderTax = {
   id: TaxType;
   percentage: number;
   taxedCountry: string;
   taxerCountry: string;
 };
 
-export interface OrderModelInterface
-  extends Model<
-    InferAttributes<OrderModelInterface, { omit: 'info' }>,
-    InferCreationAttributes<OrderModelInterface, { omit: 'info' }>
-  > {
-  id: number;
-
-  CreatedByUserId: number;
-  createdByUser?: User;
-
-  FromCollectiveId: number;
-  fromCollective?: Collective;
-  getFromCollective: HasOneGetAssociationMixin<Collective>;
-
-  CollectiveId: number;
-  collective?: Collective;
-  getCollective: HasOneGetAssociationMixin<Collective>;
-
-  TierId: number;
-  /** @deprecated: We're using both `tier` and `Tier` depending on the places. The association is defined as `Tier` (uppercase). We should consolidate to one or the other. */
-  tier?: Tier;
-  /** @deprecated: We're using both `tier` and `Tier` depending on the places. The association is defined as `Tier` (uppercase). We should consolidate to one or the other. */
-  Tier?: Tier;
-  getTier: Promise<Tier>;
-
-  quantity: number;
-  currency: string;
-  tags: string[];
-  totalAmount: number;
-  platformTipAmount: number;
-  platformTipEligible?: boolean;
-  taxAmount: number;
-  description: string;
-  publicMessage: string;
-  privateMessage: string;
-
-  SubscriptionId?: number;
-  Subscription?: SubscriptionInterface;
-  getSubscription: HasOneGetAssociationMixin<SubscriptionInterface>;
-
-  AccountingCategoryId?: ForeignKey<AccountingCategory['id']>;
-  accountingCategory?: AccountingCategory;
-  getAccountingCategory: HasOneGetAssociationMixin<AccountingCategory>;
-
-  PaymentMethodId: number;
-  paymentMethod?: PaymentMethodModelInterface;
-  getPaymentMethod: HasOneGetAssociationMixin<PaymentMethodModelInterface>;
-
-  Transactions?: TransactionInterface[];
-  getTransactions: HasManyGetAssociationsMixin<TransactionInterface>;
-
-  processedAt: Date;
-  status: OrderStatus;
-  interval?: string;
-  data:
+type Subscription = SubscriptionInterface;
+
+type AccountingCategoryId = AccountingCategory['id'];
+type ActivityId = Activity['id'];
+type CollectiveId = Collective['id'];
+type CommentId = Comment['id'];
+// type PaymentMethodId = PaymentMethodModelInterface['id'];
```

Comment:
Why comment this one, and `declare setPaymentMethod: Sequelize.BelongsToSetAssociationMixin<PaymentMethodModelInterface, PaymentMethodId>;` below?